Userland/Services/SystemServer/Service.h

```diff
@@ -60,10 +60,12 @@ class Service final : public Core::Object {
     int m_priority { 1 };
     // Whether we should re-launch it if it exits.
     bool m_keep_alive { false };
-    // Path to the socket to create and listen on on behalf of this service.
-    String m_socket_path;
-    // File system permissions for the socket.
-    mode_t m_socket_permissions { 0 };
+    // A list of each socket path that is required by the service.
+    Vector<String> m_socket_paths;
+    // File descriptors for each socket.
+    Vector<int> m_socket_fds;
+    // File system permissions for each socket.
+    Vector<mode_t> m_socket_permissions;
```

Comment:
Seems like we could bundle three socket attributes in a struct instead of having three separate vectors.

Reply:
What would be a good place to put this? Just above the `Service`?

Reply:
Sure, or as a private nested struct within `Service` if you don't need it outside the class.